llvm/include/llvm/Demangle/ItaniumDemangle.h

```diff
@@ -5737,6 +5737,7 @@ Node *AbstractManglingParser<Derived, Alloc>::parseTemplateParamDecl(
   }
 
   if (consumeIf("Tk")) {
+    ScopedOverride<bool> SaveInConstraintExpr(InConstraintExpr, true);
```

Comment:
Could you add a test that triggers this behavior?

Reply:
Yes, sure, added.